signing/verifier.go

```diff
@@ -0,0 +1,32 @@
+package signing
+
+import (
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/spacemeshos/ed25519"
+)
+
+type Verifier interface {
+	Verify(data []byte, sig []byte) (bool, error)
+	Bytes() []byte
+	String() string
+}
+
+type edVerifier struct {
+	pubKey []byte
+}
+
+func NewVerifier(pub []byte) (*edVerifier, error) {
+	return &edVerifier{pubKey: pub}, nil
+}
+
+func (ev *edVerifier) Verify(data []byte, sig []byte) (bool, error) {
+	return ed25519.Verify2(ev.pubKey, data, sig), nil
```

Comment:
I don't understand- `ev.pubKey` was extracted from the signature, right? And here you're using it to verify the signature, it seems that it will always return true, don't you think?

Reply:
extract public key will give you the extracted public key even if the sig does not much. The only way to really verify it is by matching the extracted public key with the identities you already know.

With that being said - we currently do not use verify.

I use the Verifier interface to preserve compatibility, for now, it is mainly used as a maatuf to the publickey. It is not clear to me at this point how we are gonna use it so I keep it open for now and we will see later on.

Reply:
The first paragraph is irrelevant - `FetchBLS' provides this functionality.

We may allow an auxiliary input for messages, so the public key may be sent. In this case, it is better to run verify than extract.